include/IECore/StringAlgo.h

```diff
@@ -92,6 +97,60 @@ IECORE_API bool match( const std::vector<InternedString> &path, const MatchPatte
 /// `tokenize()`, but with special handling for the "..." match token when separator is '.'.
 IECORE_API MatchPatternPath matchPatternPath( const std::string &patternPath, char separator = '/' );
 
+/// Variable Substitutions
+/// ======================
+
+/// Enum to specify what sort of substitutions may
+/// be performed on a string.
+enum Substitutions
```

Comment:
Any reason not to use `enum class`?

Reply:
A couple :

- The transition from Gaffer to Cortex is a lot easier if it doesn't change. Backwards compatibility for serialisations can be provided easily with `Gaffer.Context.Substitutions = IECore.StringAlgo.Substitutions` and likewise other code changes are trivial.
- We intentionally use the enum value to form bitmasks, and that's a bunch more faff with `enum class`. Arguably there's a case for something like [QFlags](https://doc.qt.io/qt-5/qflags.html) in Cortex, but that's out of scope for this PR.